### ISSUE TITLE
Pull `megaqc_url` with front-end javascript

### ISSUE THREAD

**Issue by tsnowlan** [ewels/MegaQC#45]
Something we talked about during the hackathon that I just remembered.

When viewing `users/multiqc_config`, the hostname and port are the local values. These won't match up if MegaQC is running in Docker or if there's apache/nginx running in front of gunicorn, etc. The easiest fix we came up with was to have the browser javascript pull the info and display it that way.

**Comment by ewels** [comment]
True, though since then I came across [the documentation](https://github.com/ewels/MegaQC/blob/master/docs/installation-prod.md#2-export-environment-variables) again:

> Add the following lines to your .bashrc file:
> 
> ```bash
> export MEGAQC_PRODUCTION=1
> export SERVER_NAME='http://megaqc.yourdomain.com'
> ```

I'm not sure that this actually _does_ anything currently - but I guess that the intention was to handle this problem. Presumably this is something that could be passed on to the docker environment fairly easily, then we could just use this in the internal flask code if given?

This may be a nicer / less hacky method.

**Comment by tsnowlan** [comment]
Haven't seen where if it pulls the env variable in somewhere, though there is an option to set it from the config.

The js method is kinda hacky, and wouldn't support a non-browser API call, but doesn't require human action. Config / env variable requires human action, but is more stable. Trade offs. :man_shrugging:

**Comment by ewels** [comment]
It [doesn't look like](https://github.com/ewels/MegaQC/search?utf8=%E2%9C%93&q=SERVER_NAME&type=) it pulls it in anywhere. I committed a bunch of super old edited files a little while back, so I suspect that this is a leftover of a half written feature. My bad... 🙄 

I figure this isn't _much_ human action - just specifying the URL (which you much already know) when you launch the container, right? 

I feel like there _must_ be a proper way to do this with Flask that I haven't found yet. It must be a super common problem...

**Comment by tsnowlan** [comment]
Aha. `SERVER_NAME` is a special flask config value: http://flask.pocoo.org/docs/0.12/config/ (have to scroll down a little). Once `SERVER_NAME` is set you can use  `url_for(...)` to generate it appropriately.

**Comment by ewels** [comment]
Aha, nice! Go past-me! So does this work with Docker? We're [already using](https://github.com/ewels/MegaQC/blob/master/megaqc/templates/users/multiqc_config.html#L16) `url_for` everywhere I think..

**Comment by tsnowlan** [comment]
It works if you've got a custom entry in your `/etc/hosts` file that points `SERVER_NAME` to the local IP. Otherwise, back to the same problem of determining external IP from inside the container. A bit more involved, but if we want to rely on `SERVER_NAME` I think the best way would be to have [docker specify an IP](https://stackoverflow.com/questions/27937185/assign-static-ip-to-docker-container) and use that.

**Comment by tsnowlan** [comment]
I think the `docker network` thing was overkill, especially since I forgot we're mostly access just by hitting localhost anyway.

The problem with setting that url as SERVER_NAME is that there's no DNS service pointing that hostname to the local docker container. We could do `-e SERVER_NAME=localhost:80` and that might work. There are some caveats on using localhost with SERVER_NAME, but they're more about using subdomains which I don't think will be an issue for us. At least for the specific use case of a local docker env.

New fun hiccup: it looks like gunicorn isn't passing `SERVER_NAME` along to the flask app. It's there in docker shell, but `url_for(..., _external=True)` is still just returning `127.0.0.1:8000`.
